application/__tests__/example.macro.integration.js

```diff
@@ -8,21 +8,47 @@ describe('Example page rendering', () => {
     await visit('/hmrc-design-patterns/ask-for-consent/')
   })
 
-  test('should have the correct meta title', async () => {
+  it('should have the correct meta title', async () => {
     const title = await page.title()
     expect(title).toBe('Ask the user for their consent - HMRC Design Patterns - Design resources for HMRC – GOV.UK')
   })
 
-  test('iframes should have a style attribute with a height property', async () => {
-    const noHeightProperty = await page.$$eval('iframe', iframes => iframes.filter(iframe => !iframe.style.height))
-    expect(noHeightProperty).toHaveLength(0)
+  describe('iFrame resizer', () => {
+    it('should automatically set a height property on all iframes', async () => {
+      const noHeightProperty = await page.$$eval('iframe', iframes => iframes.filter(iframe => !iframe.style.height))
+      expect(noHeightProperty).toHaveLength(0)
+    })
+
+    it('should automatically size iframes to ensure all content is displayed', async () => {
+      const resizeFailed = await page.$$eval('iframe', (iframes) => {
+        const isFrameATardis = iframe => iframe.offsetHeight < iframe.contentWindow.document.body.offsetHeight
+        return iframes.filter(isFrameATardis)
+      })
+      expect(resizeFailed).toHaveLength(0)
+    })
   })
 
-  test('iframes should be resized to display all content', async () => {
-    const resizeFailed = await page.$$eval('iframe', (iframes) => {
-      const isFrameATardis = iframe => iframe.offsetHeight < iframe.contentWindow.document.body.offsetHeight
-      return iframes.filter(isFrameATardis)
+  describe('Copy snippet button', () => {
+    const buttonSelector = '#example-yes-no-question-html .app-link--copy';
+
+    beforeEach(async () => {
+      await page.click('.example-example-yes-no-question-lang ul > li:first-child > a')
+      await page.click(buttonSelector)
     })
-    expect(resizeFailed).toHaveLength(0)
+
+    it('should change the button text to `copied` when clicked', async () => {
+      const buttonText = await page.$eval(buttonSelector, btn => btn.textContent)
+      expect(buttonText).toBe('Copied')
+    })
+
+    it('should reset text to `Copy` after an interval', async () => {
+      let buttonText
+      await new Promise(resolve => setTimeout(async () => {
+        buttonText = await page.$eval(buttonSelector, btn => btn.textContent)
+        resolve()
+      }, 5000))
```

Comment:
We shouldn't put real delays into tests, this should be mockable :)

Reply:
AS this is an integration test mocking a delay is overly complex and probably not desirable